src/Http/Controllers/Api/UserDiskController.php

```diff
@@ -35,21 +40,33 @@ class UserDiskController extends Controller
      */
     public function store(StoreUserDisk $request)
     {
-        $disk = UserDisk::create([
-            'name' => $request->input('name'),
-            'type' => $request->input('type'),
-            'user_id' => $request->user()->id,
-            'expires_at' => now()->addMonths(config('user_disks.expires_months')),
-            'options' => $request->getDiskOptions(),
-        ]);
-
-        if ($this->isAutomatedRequest()) {
-            return $disk;
-        }
+        try {
+            $disk = DB::transaction(function () use ($request) {
+                $disk = UserDisk::create([
+                    'name' => $request->input('name'),
+                    'type' => $request->input('type'),
+                    'user_id' => $request->user()->id,
+                    'expires_at' => now()->addMonths(config('user_disks.expires_months')),
+                    'options' => $request->getDiskOptions(),
+                ]);
+
+                $this->validateS3Config($disk);
+
+                return $disk;
+            });
 
-        return $this->fuzzyRedirect('storage-disks')
-            ->with('message', 'Storage disk created')
-            ->with('messageType', 'success');
+            if ($this->isAutomatedRequest()) {
+                return $disk;
+            }
+
+            return $this->fuzzyRedirect('storage-disks')
+                ->with('message', 'Storage disk created')
+                ->with('messageType', 'success');
+        } catch (ValidationException $e) {
+            return $this->fuzzyRedirect()
+                ->withErrors($e->errors())
+                ->withInput();
+        }
```

Comment:
No need for the try/catch here. The ValidationException handles both browser and script requests, so there is no need to call `fuzzyRedirect()`. This is only needed in the success state (as it was before).